src/concepts/components/internalstate.md

```diff
@@ -0,0 +1,169 @@
+---
+description: Component can maintain it's own state and render information depending on it
+---
+
+# Internal State
+
+The component can manage it's own state using the Rust struct that implement the trait `Component`. The HTML rendering is based on this state.
```

Comment:
"manage it's own state" => "manage its own state"
I think that the struct is a Rust struct is clear from the context so Rust can be removed here. Also, struct shouldn't use the definite article as a multitude of potential structs are being referred to, rather than a single struct (i.e. `the Rust struct` => `a struct`)